src/grid.scss

```diff
@@ -0,0 +1,119 @@
+@import "./mixins/breakpoints.scss";
+@import "./variables/grid.scss";
+
+/*
+ * Grid classes
+ * Component code   : gd
+ *
+ * Modifier 1       : gap   ( grid gap modifier )
+ *                    col   ( grid column modifier )
+ *                    cols  ( grid columns template modifier )
+ *
+ * Modifier 2       : eq ( grid with equal sized cols ) [Modifier 1: cols]
+
+ *                      Variables used:
+ *                          --gd-cols-eq     ( num cols )
+ *                          --gd-cols-eq-m   ( num cols @m )
+ *                          --gd-cols-eq-l   ( num cols @l )
+ *                          --gd-cols-eq-xl  ( num cols @xl )
+ *                          --gd-cols-eq-2xl ( num cols @2xl )
+ *
+ *                    cus ( grid with custom cols size ) [Modifier 1: cols,col,gap]
+ *
+ *                      Variables used:
+ *                        [Modifier 1: cols]
+ *                          --gd-cols-cus     (num of cus cols )
+ *                          --gd-cols-cus-m   (num of cus cols @m )
+ *                          --gd-cols-cus-l   (num of cus cols @l )
+ *                          --gd-cols-cus-xl  (num of cus cols @xl )
+ *                          --gd-cols-cus-2xl (num of cus cols @2xl )
+ *
+ *                        [Modifier 1: gap]
+ *                          --gd-gap-cus     ( gap size s )
+ *                          --gd-gap-cus-m   ( gap size m )
+ *                          --gd-gap-cus-l   ( gap size l )
+ *                          --gd-gap-cus-xl  ( gap size xl )
+ *                          --gd-gap-cus-2xl ( gap size 2xl )
+ *
+ *                    subgrid (subgrid within col)
+ *
+ *                      Variables used:
+ *                        [Modifier 1: col]
+ *                          --gd-col-cus ( for cus subgrid within a col )
+ *
+ */
+
+.gd-gap-cus-m { gap: $grid-m; }
+.gd-gap-cus-l { gap: $grid-l; }
+.gd-gap-cus-xl { gap: $grid-xl; }
+.gd-gap-cus-2xl { gap: $grid-2xl; }
+.gd-gap-cus { gap: var(--gd-gap-cus, repeat(12, 1fr)); }
+
+@mixin gd {
+    display: grid;
+}
+@mixin gap($row, $col) {
+    row-gap: $row;
+    column-gap: $col;
+}
+
+.gd-cols-eq { @include gd;
+              grid-template-columns: repeat(var(--gd-cols-eq, 12), 1fr);
+}
+.gd-cols-cus { @include gd;
+               grid-template-columns: var(--gd-cols-cus, repeat(12, 1fr));
+}
+.gd-col-subgrid { @include gd;
+                  grid-template-columns: subgrid;
+                  grid-column: var(--gd-col-cus, repeat(12, 1fr));
```

Comment:
I think the default value for this is not right. This default value will be good for grid-template-columns. Instead we should have `span 1` as the default